src/dnx.mono.managed/EntryPoint.cs

```diff
@@ -69,82 +68,92 @@ public static int Main(string[] arguments)
             Environment.SetEnvironmentVariable(EnvironmentNames.AppBase, arguments[appbaseIndex + 1]);
         }
 
-        return RuntimeBootstrapper.Execute(arguments, 
+        return RuntimeBootstrapper.Execute(arguments,
             // NOTE(anurse): Mono is always "dnx451" (for now).
             new FrameworkName("DNX", new Version(4, 5, 1)));
     }
 
     private static string[] ExpandCommandLineArguments(string[] arguments)
     {
-        // If '--appbase' is already given and it has a value, don't need to expand
-        var appbaseIndex = arguments.ToList().FindIndex(arg =>
-            string.Equals(arg, "--appbase", StringComparison.OrdinalIgnoreCase));
-        if (appbaseIndex >= 0 && (appbaseIndex < arguments.Length - 1))
+        int parameter_idx = FindAppBaseOrNonHostOption(arguments);
+
+        // no non-bootstrapper parameters found or --appbase was found
+        if (parameter_idx < 0 || string.Equals(arguments[parameter_idx], "--appbase", StringComparison.OrdinalIgnoreCase))
         {
             return arguments;
         }
 
         var expandedArgs = new List<string>();
-
-        // Copy all arguments (options & values) as is before the project.json/assembly path
-        var pathArgIndex = -1;
-        while (++pathArgIndex < arguments.Length)
+        for (var i = 0; i < arguments.Count(); i++)
         {
-            var optionValNum = BootstrapperOptionValueNum(arguments[pathArgIndex]);
-
-            // It isn't a bootstrapper option, we treat it as the project.json/assembly path
-            if (optionValNum < 0)
+            if (i == parameter_idx)
             {
-                break;
+                ExpandArgument(arguments[i], expandedArgs);
             }
-
-            // Copy the option
-            expandedArgs.Add(arguments[pathArgIndex]);
-
-            // Copy the value if the option has one
-            if (optionValNum > 0 && (++pathArgIndex < arguments.Length))
+            else
             {
-                expandedArgs.Add(arguments[pathArgIndex]);
+                expandedArgs.Add(arguments[i]);
             }
         }
 
-        // No path argument was found, no expansion is needed
-        if (pathArgIndex >= arguments.Length)
-        {
-            return arguments;
-        }
+        return expandedArgs.ToArray();
```

Comment:
`Bootstrapper.Execute` takes parameters as an array of strings so we have to allocate it somewhere...